Tweak `alternating_group` and `symmetric_group`

fingolfin opened the pull request:
- improve the docstrings, add examples
- remove obscure ability to request a `PcGroup` instead of a `PermGroup`, which only worked with degrees <= 4 anyway; if one needs these groups one can easily get them via e.g. `PcGroup(alternating_group(4))` which to me is conceptually much clearer (and performance does not matter as this is only for interactive toy code anyway)

ThomasBreuer commented on src/Groups/GAPGroups.jl:
```diff
@@ -182,8 +182,8 @@ end
-@gapwrap has_order(G::GAPGroup) = GAP.Globals.HasSize(G.X)
-@gapwrap set_order(G::GAPGroup, val::T) where T<:IntegerUnion = GAP.Globals.SetSize(G.X, GapObj(val))
+@gapwrap has_order(G::GAPGroup) = GAP.Globals.HasOrder(G.X)
+@gapwrap set_order(G::GAPGroup, val::T) where T<:IntegerUnion = GAP.Globals.SetOrder(G.X, GapObj(val))
```
I do not understand this.
In GAP, `Size` and `Order` are independent operations, as well as their testers and setters.
If `HasSize( G )` is `true` then `HasOrder( G )` can be `false`, and calling `SetOrder` for a group does not set `Size` for it. Since `Size` is the attribute used for groups, `HasOrder` and `SetOrder` do not help.

fingolfin replied:
Oops, thanks for catching this. (It's from a separate commit that I meant to discard)

ThomasBreuer reviewed (commented):
Two more tests use the no longer supported type argument.
I do not understand the switch from calling `GAP.Globals.HasSize` to `GAP.Globals.HasOrder`.

fingolfin commented:
The broken tests are due to me trying to untangle parts of an old branch of mine; there is another upcoming PR with the changes required to make this PR and PR #1574 pass, I mistakenly assumed they were independent *sigh*

fingolfin committed:
Tweak {alternating,symmetric}_group

- improve the docstrings, add examples
- remove obscure ability to request a `PcGroup` instead of a `PermGroup`,
  which only worked with degrees <= 4 anyway; if ones need these groups,
  one can easily get them via e.g. `PcGroup(alternating_group(4))`

fingolfin commented:
Actually this one indeed just needed two tests to be adjusted. And I dropped the bad commit

ThomasBreuer reviewed (approved):
Looks good to me. Thanks.
